src/source/Ice/IceAgent.c

```diff
@@ -89,7 +89,19 @@ STATUS createIceAgent(PCHAR username, PCHAR password, PIceAgentCallbacks pIceAge
                                             (PCHAR) pRtcConfiguration->iceServers[i].urls,
                                             (PCHAR) pRtcConfiguration->iceServers[i].username,
                                             (PCHAR) pRtcConfiguration->iceServers[i].credential))) {
+            pIceAgent->rtcIceServerDiagnostics[i].port = (INT32)getInt16(pIceAgent->iceServers[i].ipAddress.port);
+            if(pIceAgent->iceServers[pIceAgent->iceServersCount].transport == KVS_SOCKET_PROTOCOL_UDP) {
+                STRCPY(pIceAgent->rtcIceServerDiagnostics[i].protocol, ICE_URL_TRANSPORT_UDP);
+            }
+            else if(pIceAgent->iceServers[pIceAgent->iceServersCount].transport == KVS_SOCKET_PROTOCOL_TCP) {
+                STRCPY(pIceAgent->rtcIceServerDiagnostics[i].protocol, ICE_URL_TRANSPORT_TCP);
+            }
+            else {
+                MEMSET(pIceAgent->rtcIceServerDiagnostics[i].protocol, '\0', SIZEOF(pIceAgent->rtcIceServerDiagnostics[i].url));
+            }
+            STRCPY(pIceAgent->rtcIceServerDiagnostics[i].url, pRtcConfiguration->iceServers[i].urls);
             pIceAgent->iceServersCount++;
+            CHK_STATUS(hashTableCreateWithParams(ICE_HASH_TABLE_BUCKET_COUNT, ICE_HASH_TABLE_BUCKET_LENGTH, &pIceAgent->rtcIceServerDiagnostics[i].requestTimestampDiagnostics));
```

Comment:
Important! You don't need separate hash tables. Create one for all ice agents as you are using CRC32 as the key and you don't have enumeration operation to list all of the keys for a single ice agent